tests/phpunit/TestFramework/Coverage/XmlReport/IndexXmlCoverageLocatorTest.php

```diff
@@ -0,0 +1,166 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\TestFramework\Coverage\XmlReport;
+
+use const DIRECTORY_SEPARATOR;
+use Infection\FileSystem\Locator\FileNotFound;
+use Infection\TestFramework\Coverage\XmlReport\IndexXmlCoverageLocator;
+use Infection\Tests\FileSystem\FileSystemTestCase;
+use function Infection\Tests\normalizePath;
+use const PHP_OS_FAMILY;
+use function Safe\chdir;
+use function Safe\mkdir;
+use function Safe\realpath;
+use function Safe\sprintf;
+use function Safe\touch;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * @group integration
+ */
+final class IndexXmlCoverageLocatorTest extends FileSystemTestCase
+{
+    /**
+     * @var IndexXmlCoverageLocator
+     */
+    private $locator;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // Move to the temporary directory: we want to make sure the setUp closures are executed
+        // there since they do not have access to the tmp yet, so their paths are relative
+        chdir($this->tmp);
+        mkdir($this->tmp . '/coverage-xml');
+
+        $this->locator = new IndexXmlCoverageLocator($this->tmp . '/coverage-xml');
+    }
+
+    protected function tearDown(): void
+    {
+        chdir($this->cwd);
+
+        parent::tearDown();
+    }
+
+    public function test_it_can_locate_the_default_index_file(): void
+    {
+        touch('coverage-xml/index.xml');
+
+        $expected = normalizePath(realpath($this->tmp . '/coverage-xml/index.xml'));
+
+        $this->assertSame($expected, $this->locator->locate());
+        // Call second time to check the cached result
+        $this->assertSame($expected, $this->locator->locate());
+    }
+
+    public function test_it_can_locate_the_default_index_file_with_the_wrong_case(): void
+    {
+        if (PHP_OS_FAMILY !== 'Darwin') {
+            $this->markTestSkipped('Cannot test this on case-sensitive OS');
+        }
+
+        touch('coverage-xml/INDEX.XML');
+
+        $expected = normalizePath(realpath($this->tmp . '/coverage-xml/index.xml'));
+
+        $actual = $this->locator->locate();
+
+        $this->assertSame($expected, $actual);
+    }
+
+    /**
+     * @dataProvider indexPathsProvider
+     */
+    public function test_it_can_find_more_exotic_index_file_names(string $indexRelativePath): void
+    {
+        (new Filesystem())->dumpFile($indexRelativePath, '');
+
+        $expected = normalizePath(realpath($this->tmp . DIRECTORY_SEPARATOR . $indexRelativePath));
+
+        $this->assertSame($expected, $this->locator->locate());
+        // Call second time to check the cached result
+        $this->assertSame($expected, $this->locator->locate());
+    }
+
+    public function test_it_cannot_locate_the_index_file_if_the_result_is_ambiguous(): void
+    {
+        touch('index.xml');
```

Comment:
didn't get this case. Shouldn't this loginc from locator immediately return `index.xml` and do not search in nested folders:

```php
if (file_exists($this->defaultIndexPath)) {
    return $this->indexPath = $this->defaultIndexPath;
}
```

Reply:
the default path is `coverage/coverage-xml/index.xml`, here it is respectively `coverage/index.xml` and `coverage/sub-dir/index.xml`